miden-lib/asm/miden/contracts/wallets/basic.masm

```diff
@@ -53,9 +57,18 @@ export.send_asset
     # is because the VM stack has a minimum size of 16 elements, trying to push
     # elements after the call to `create_note` would increase the stack in
     # addition to the minimum 16 elements.
-    push.0.0 padw movdnw.3 padw movdnw.3 movdn.11 drop
-    # => [ASSET, tag, note_type, RECIPIENT, ZERO, ZERO, 0, ...]
+    push.0 movdn.6 push.0 movdn.6 padw movdnw.2 padw movdnw.2
+    # => [tag, note_type, RECIPIENT, 0, 0, ZERO, ZERO, ...]
 
     exec.tx::create_note
-    # => [note_ptr, ZERO, ZERO, 0, ...]
+    # => [note_ptr, 0, ZERO, ZERO, ...]
+
+    padw mem_loadw.0 movup.4
+    # => [note_ptr, ASSET, 0, ZERO, ZERO, ...]
+
+    exec.tx::add_asset_to_note
+    # => [note_ptr, 0, ZERO, ZERO, ...]
+
+    # prepare the stack for return - stack has 5 elements too many
+    movupw.3 dropw swap drop
```

Comment:
Why do we need to do this now while previously we didn't have to?

Separately, I wonder if a better approach here would be to call `sys::truncate_stack`. It would be a bit more expensive - but also generic in a way we won't need to worry about how many elements on the staack are extra. This could be done as a part of #685.